File: api-gin/internal/models/menu.go
package models

import (
	"time"
)

type MenuModel struct {
	Conn  string
	Table string
}

var MenuMdl = &MenuModel{"default", "menu"}

type MenuRow struct {
	ID          int64     `db:"id"`
	Name        string    `db:"name"`   //  菜单名称
	Uri         string    `db:"uri"`    //  uri地址
	Level       int64     `db:"level"`  //  节点等级
	Pid         int64     `db:"pid"`    //  父节点菜单id
	Icon        string    `db:"icon"`   //  菜单图标
	Status      int64     `db:"status"` //  状态（0-禁用，1-正常）
	Sort        int64     `db:"sort"`   //  排序
	Remark      string    `db:"remark"`
	CreatedTime time.Time `db:"created_time"`
	UpdatedTime time.Time `db:"updated_time"`
}

var MENU_STATUS_MAP = map[string]int{
	"valid":   1,
	"invalid": 0,
}

File: api-fk/app/Models/Admin/Menu.php
<?php

namespace App\Models\Admin;

use App\Models\Model;

use App\Components\Helper;

class Menu extends Model
{
    protected $connection = 'default';
    protected $table = 'menu';
    protected $fields = [
        'id' => 'int',
        'name' => 'string',
        'uri' => 'string',
        'level' => 'int',
        'pid' => 'int',
        'icon' => 'string',
        'status' => 'int',
        'sort' => 'int',
        'remark' => 'string',
        'created_time' => 'string',
        'updated_time' => 'string',
    ];

    const MAP_STSTUA = [
        'valid' => 1,
        'invalid' => 0,
    ];

    public function getAllMenu(bool $tree = true)
    {
        $list = $this->take();
        return $tree ? $this->_genTree($list) : $list;
    }

    public function getValidAllMenu(bool $tree = true)
    {
        $list = $this->take(['status' => self::MAP_STSTUA['valid']]);
        return $tree ? $this->_genTree($list) : $list;
    }

    public function getTree(array $mids): array
    {
        if (!$mids) {
            return [];
        }
        $list = $this->take(['id' => $mids, 'status' => self::MAP_STSTUA['valid']]);
        return $this->_genTree($list);
    }

    private function _genTree(array $list = []): array
    {
        if (!$list) {
            return [];
        }
        return Helper::genTree($list, 'id', 'pid', 'sub_menu');
    }
}
